src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -21,6 +21,7 @@ public class Connection: IConnection
         private ITransport _transport;
         private Task _receiveLoopTask;
         private volatile Task _startTask = Task.CompletedTask;
+        private volatile Task _eventQueue = Task.CompletedTask;
```

Comment:
I can try but it had some things I was not sure about (like TaskMonitor to warn about possible deadlocks). It would also have to be rewritten/cleaned up a bit because it has references to performance counters and TaskAsyncHelper. If the tests can be ported easily then maybe it's fine.